### PR TITLE
Find image from tags

### PR DESCRIPTION
**Pull request by ajordens**

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    String cloudProvider = getCloudProvider(stage);
+    ImageFinder imageFinder = imageFinders.stream()
+      .filter(it -> it.getCloudProvider().equals(cloudProvider))
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException("ImageFinder not found for cloudProvider " + cloudProvider));
+
+    StageData stageData = (StageData) stage.mapTo(StageData.class);
+    return new DefaultTaskResult(
+      ExecutionStatus.SUCCEEDED,
+      Collections.singletonMap("amiDetails", imageFinder.byTags(stage, stageData.packageName, stageData.tags))
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java]: Explicitly did not put anything in the _global context_ ... but did verify that this hangs together with the `DeploymentDetailsAware` lookup (_ie. can do a `FindImageFromTags` followed by a `Deploy`_).

At one point I believe @duftler was going to get `DeploymentDetailsAware` to span pipeline executions.

**Comment by ajordens** [comment]
PTAL @spinnaker/netflix-reviewers (FYI @spinnaker/google-reviewers)

**Comment by anotherchrisberry** [comment]
LGTM

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java (hunk 1 above)
Comment (duftler) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/FindImageFromTagsTask.java]: Yes, still planning on doing that work. The recent collection of PR's are all related to things that came up as I've tried to work through it.

**Pull request merged**